---
string[] inputs
string[] outputs
bool success